app/scripts/controllers/cached-balances.js

```diff
@@ -0,0 +1,83 @@
+const ObservableStore = require('obs-store')
+const extend = require('xtend')
+
+/**
+ * @typedef {Object} CachedBalancesOptions
+ * @property {Object} accountTracker Account tracker store reference
+ * @property {Function} getNetwork Returns the current network
+ * @property {Object} initState Initial state to with which to populate
+ */
+
+/**
+ * Background controller responsible for maintaining
+ * a cache of account balances in local storage
+ */
+class CachedBalancesController {
+  /**
+   * Creates a new controller instance
+   *
+   * @param {CachedBalancesOptions} [opts] Controller configuration parameters
+   */
+  constructor (opts = {}) {
+    const { accountTracker, getNetwork } = opts
+
+    this.accountTracker = accountTracker
+    this.getNetwork = getNetwork
+
+    const initState = extend({
+      cachedBalances: {},
+    }, opts.initState)
+    this.store = new ObservableStore(initState)
+
+    this._registerUpdates()
+  }
+
+  /**
+   * Updates the cachedBalances property for the current network. Cached balances will be updated to those in the passed accounts
+   * if balances in the passed accounts are truthy.
+   *
+   * @param {accounts} [object] Contains the recently updated accounts for the current network
```

Comment:
```suggestion
   * @param {Object} obj The the recently updated accounts object for the current network
```

Reply:
The type and param name here are mixed up. I've added a suggestion above but feel free to use something else.